### ISSUE TITLE
Model selection

### ISSUE THREAD

**Issue by peter943** [Wensi-Tang/OS-CNN#3]
Hi,
I have learnt your paper these day, which is very enlightening to me. I have a question about model selection.  I do not know which model (The model will be saved many times during the training) to choose to reproduce your results. Could you please give me some help?

**Comment by Wensi-Tang** [comment]
Hi,

Thanks for your attention. I hope my answer could help you

**Q**:  Which model (The model will be saved many times during the training) to choose to reproduce your results?

**A:**  For 'XX' dataset, the model named "XXBest_model" is what you want.

Using [this](https://github.com/Wensi-Tang/OS-CNN/tree/master/Example_Results_of_OS_CNN/OS_CNN_result_iter_0/FiftyWords) result of "FiftyWords" datasets as an example, the model 'FiftyWordsBest_model' is the model you want.

*************************************



Other information might help you.

1)  The link [here ](https://github.com/Wensi-Tang/OS-CNN/blob/master/1_2_OS-CNN_load_saved_model_for_prediction.ipynb) is an example to load the model and use it for prediction.


2) Why do I save the model for many times?

 I use this as a checkpoint. 

The frequency of model saving depends on a hyperparameter `print_result_every_x_epoch` in this [file](https://github.com/Wensi-Tang/OS-CNN/blob/master/1_1_1_OS-CNN_easy_use_Run_and_Save_Model_for_multivariate.ipynb)

the default value is 
`print_result_every_x_epoch = 50`
which means it will save the model every 50 epoch.

But no need to worry about this setting
No matter the `max_epoch`  is an integer multiple of `print_result_every_x_epoch` or not, the model will be saved after the last epoch finish. You can check it [here](https://github.com/Wensi-Tang/OS-CNN/blob/master/Classifiers/OS_CNN/OS_CNN_easy_use.py#L131)


Thanks for you attention again, if you have any further questions feel free to tell me.

**Comment by peter943** [comment]
@Wensi-Tang Thanks for your kind help. 
The results you get in the Result_of_each_iter_for_128_datasets. csv are tested using the last saved model according to the your code, this model may not be the model which has minimum training loss or maximum test accuracy. I do not know this understanding is right?

**Comment by Wensi-Tang** [comment]
Yea. I just use the model saved at 2000 epoch and don't choose the model based on training loss or test accuracy. 

I do this for simplicity, and I am not sure the performance would be higher or lower if I choose the model based on minimum training loss.

**Comment by Wensi-Tang** [comment]
@peter943 Thanks for you attention too!

Your understanding is right. I use the model saved when the 2000th epoch finishes and don't select the model based on any other thing.  


***********************
Other information might help you.

I am not sure the performance will be higher or lower if I choose the model based on minimum training loss.

**Comment by peter943** [comment]
@Wensi-Tang Ok. Thank you for your patient reply.

**Comment by Wensi-Tang** [comment]
@peter943  Thanks you too! 
Your comment inspired me to check my code and I just found the [file](https://github.com/Wensi-Tang/OS-CNN/blob/master/1_1_OS-CNN_easy_use_Run_and_Save_Model.ipynb) has some formatting problem and cannot render, I fixed it.
